File: Day 7/menstrationWahala/menstrationWahala.js
const prompt = require('prompt-sync')();

let periodStartDate;
let periodLength;
while (true) {
    let periodStartInput = prompt("When did your last period start? (YYYY-MM-DD) ");

     periodStartDate = new Date(periodStartInput);


    if (periodStartInput.includes('-') && periodStartInput.length === 10 && !isNaN(periodStartDate.getTime())) {
        console.log("Your last period started on: " + periodStartDate.toDateString());
     	break;
   		 } 

	else {
        console.log("Invalid format. Please enter the date in YYYY-MM-DD");
    		}

}


let askUserIfSheKnowsperiodLength = prompt("Do you know how long your period lasts? ")

	if (askUserIfSheKnowsperiodLength.toLowerCase() === 'yes'){

	periodLength = prompt("How long does your period last? ")

	periodLength = parseInt(periodLength);

		}

	if (askUserIfSheKnowsperiodLength.toLowerCase() === 'no'){ 
	periodLength = 5
			}


while(true){
let askUserIfSheKnowsperiodCycle = prompt("Do you know how long your cycle is? ")

	if (askUserIfSheKnowsperiodCycle !== 'yes' && askUserIfSheKnowsperiodCycle !== 'no'){
	console.log('Please enter yes or no')
		}

	if (askUserIfSheKnowsperiodCycle.toLowerCase() === 'yes'){

	let cycleLength = prompt("How long does your cycle last? ")

	cycleLength = parseInt(cycleLength);

	nextFlow =  new Date(periodStartDate);

	nextFlow.setDate(periodStartDate.getDate() + cycleLength);

	let ovulationDate = new Date(periodStartDate)

	ovulationDate.setDate(periodStartDate.getDate() + 14)

	let fertilityStart = new Date(ovulationDate)

	fertilityStart.setDate(ovulationDate.getDate() - 5)

	let fertilityEnd = new Date(ovulationDate)

	fertilityEnd.setDate(ovulationDate.getDate() + 2)

	let safePeriodStart = new Date(ovulationDate)
	
	safePeriodStart.setDate(ovulationDate.getDate() + 5)

	let safePeriodEnd = new Date(nextFlow)

	safePeriodEnd.setDate(nextFlow.getDate() - 5)

	let periodEnd = new Date(nextFlow)

	periodEnd.setDate(nextFlow.getDate() + periodLength)


	console.log("\nThe next period start date is: " + nextFlow.toDateString());
	console.log("Your ovulation period is: " + ovulationDate.toDateString())
	console.log("Your fertility period is: " + fertilityStart.toDateString()+ " to "+ fertilityEnd.toDateString());
	console.log("Your safe period is: " + safePeriodStart.toDateString()+ " to "+ safePeriodEnd.toDateString());
	console.log("Your period will last from: " + nextFlow.toDateString()+ " to "+ periodEnd.toDateString());
	
	break;
			}


		if (askUserIfSheKnowsperiodCycle.toLowerCase() === 'no'){ 
		cycleLength = 28

		nextFlow =  new Date(periodStartDate);

		nextFlow.setDate(periodStartDate.getDate() + cycleLength);
		
		ovulationDate = new Date(nextFlow)

		ovulationDate.setDate(nextFlow.getDate() - 12)

		fertilityStart = new Date(ovulationDate)

		fertilityStart.setDate(ovulationDate.getDate() - 5)

		fertilityEnd = new Date(ovulationDate)

		fertilityEnd.setDate(ovulationDate.getDate() + 2)

		safePeriodStart = new Date(ovulationDate)
	
		safePeriodStart.setDate(ovulationDate.getDate() + 5)

		safePeriodEnd = new Date(nextFlow)

		safePeriodEnd.setDate(nextFlow.getDate() - 5)

		let periodEnd = new Date(nextFlow)

		periodEnd.setDate(nextFlow.getDate() + periodLength)

		console.log("\nThe next period start date is: " + nextFlow.toDateString());
		console.log("Your ovulation period is: " + ovulationDate.toDateString());
		console.log("Your fertility period is: " + fertilityStart.toDateString()+ " to "+ fertilityEnd.toDateString());
		console.log("Your safe period is: " + safePeriodStart.toDateString()+ " to "+ safePeriodEnd.toDateString());
		console.log("Your period will last from: " + nextFlow.toDateString()+ " to "+ periodEnd.toDateString());

		break;
			}

			}

File: Day 7/menstrationWahala/menstration_wahala.py
from datetime import datetime, timedelta

def get_date_input(prompt_message):
    while True:
        date_str = input(prompt_message)
        try:
            date_obj = datetime.strptime(date_str, "%Y-%m-%d")
            return date_obj
        except ValueError:
            print("Invalid format. Please enter the date in YYYY-MM-DD")

period_start_date = get_date_input("When did your last period start? (YYYY-MM-DD) ")

print("Your last period started on:", period_start_date.strftime("%A, %B %d, %Y"))

ask_user_if_knows_period_length = input("Do you know how long your period lasts? ").lower()

if ask_user_if_knows_period_length == 'yes':
    period_length = int(input("How long does your period last? "))
else:
    period_length = 5

while True:
    ask_user_if_knows_period_cycle = input("Do you know how long your cycle is? ").lower()
    
    if ask_user_if_knows_period_cycle not in ['yes', 'no']:
        print('Please enter yes or no')
        continue
    
    if ask_user_if_knows_period_cycle == 'yes':
        cycle_length = int(input("How long does your cycle last? "))
        next_flow = period_start_date + timedelta(days=cycle_length)
        ovulation_date = period_start_date + timedelta(days=14)
    else:
        cycle_length = 28
        next_flow = period_start_date + timedelta(days=cycle_length)
        ovulation_date = next_flow - timedelta(days=12)

    fertility_start = ovulation_date - timedelta(days=5)
    fertility_end = ovulation_date + timedelta(days=2)
    safe_period_start = ovulation_date + timedelta(days=5)
    safe_period_end = next_flow - timedelta(days=5)
    period_end = next_flow + timedelta(days=period_length)

    print("\nThe next period start date is:", next_flow.strftime("%A, %B %d, %Y"))
    print("Your ovulation period is:", ovulation_date.strftime("%A, %B %d, %Y"))
    print("Your fertility period is:", fertility_start.strftime("%A, %B %d, %Y"), "to", fertility_end.strftime("%A, %B %d, %Y"))
    print("Your safe period is:", safe_period_start.strftime("%A, %B %d, %Y"), "to", safe_period_end.strftime("%A, %B %d, %Y"))
    print("Your period will last from:", next_flow.strftime("%A, %B %d, %Y"), "to", period_end.strftime("%A, %B %d, %Y"))
    break
